CONTRIBUTING.md

```diff
@@ -78,11 +78,135 @@ strategy to maintain the project. **PLEASE FOLLOW THIS FORMAT**
   a maintainer asks you to reduce your PR size, please do so.
 - `release/vX.X.X` Used to release the sprints work into production.
 
+## Commit Messages
+
+### Message Structure
+
+A commit message consists of three distinct parts separated by a blank line: the title, an **optional** body and an **optional** footer. The layout looks like this:
+
+`type(scope?): subject`
+
+`Body`
+
+`Footer`
+
+The title consists of the type of the message and subject.
+
+### :exclamation: Command Line
+
+This is how you commit your message with all three parts with a blank line:
+
+`git commit -m "type(scope)?: subject" -m "Body..." -m "Footer..."`
+
+### The Type
+The type is contained within the title and can be one of these types:
+
+- **feat**: a new feature
+- **fix**: a bug fix
+- **docs**: changes to documentation
+- **style**: formatting, missing semicolons, etc; no code change
+- **refactor**: refactoring production code
+- **test**: adding tests, refactoring test; no production code change
+- **chore**: updating build tasks, package manager configs, etc; no production code change
+
+### The Scope
+The scope describes the category of your change and the commonly used are of the following: 
+- build, 
+- chore,
+- ci,
+- docs,
+- feat,
+- fix,
+- perf,
+- refactor,
+- revert,
+- style,
+- test
+
+NOTE: These strict type and scope can be found in **commitlint.config.js** 
+
+### The Subject
+
+Subjects should be no greater than 50 characters, does not begin with a capital letter and do not end with a period.
+Use an imperative tone to describe what a commit does, rather than what it did. For example, use change; not changed or changes.
+
+### The Body
+
+Not all commits are complex enough to warrant a body, therefore it is optional and only used when a commit requires a bit of explanation and context. Use the body to explain the what and why of a commit, not the how.
+When writing a body, the blank line between the title and the body is required and you should limit the length of each line to no more than 100 characters.
+
+### The Footer
+
+The footer is optional and is used to reference issue tracker IDs.
+
```

Comment:
Same here, we can remove these. Issue reference can be put in the subject. EG: `ci: fixes #44`

Reply:
Sure. No problem :)